src/amoc_config/amoc_config_attributes.erl

```diff
@@ -0,0 +1,167 @@
+    RequiredVariables = proplists:get_all_values(AttrName, ModuleAttributes),
+    lists:append(RequiredVariables).
+
+-spec process_module_attributes(module(), [maybe_module_attribute()]) ->
+    {ok, module_configuration()} | amoc_config_utils:error().
+process_module_attributes(Module, ScenarioAttributes) ->
+    Config = [process_var_attr(Module, Attr)
+              || Attr <- ScenarioAttributes],
+    amoc_config_utils:maybe_error(invalid_attribute_format, Config).
+
+-spec process_var_attr(module(), maybe_module_attribute()) ->
+    {ok, module_parameter()} | {error, reason()}.
+process_var_attr(Module, Attr) ->
+    PipelineActions = [
+        {fun check_mandatory_fields/1, []},
+        {fun check_default_value/1, []},
+        {fun check_verification_method/1, []},
+        {fun check_update_method/1, []},
+        {fun make_module_parameter/2, [Module]}],
+    case amoc_config_utils:pipeline(PipelineActions, {ok, Attr}) of
+        {error, Reason} -> {error, add_original_attribute(Reason, Attr)};
+        {ok, Param} -> {ok, Param}
+    end.
+
+-spec check_mandatory_fields(maybe_module_attribute()) ->
+    {ok, maybe_module_attribute()} | {error, reason()}.
+check_mandatory_fields(#{description := List, name := Atom} = Attr) when is_atom(Atom),
+                                                                         is_list(List) ->
+    case io_lib:char_list(List) of
+        true -> {ok, Attr};
+        false -> {error, invalid_attribute}
+    end;
+check_mandatory_fields(_Attr) ->
+    {error, invalid_attribute}.
+
+-spec check_default_value(maybe_module_attribute()) -> {ok, maybe_module_attribute()}.
+check_default_value(Attr) ->
+    DefaultValue = maps:get(default_value, Attr, undefined),
+    {ok, Attr#{default_value => DefaultValue}}.
+
+-spec check_verification_method(maybe_module_attribute()) ->
+    {ok, maybe_module_attribute()} | {error, reason()}.
+check_verification_method(Attr) ->
+    VerificationMethod = maps:get(verification, Attr, none),
+    case verification_fn(VerificationMethod) of
+        not_exported ->
+            {error, verification_method_not_exported};
+        invalid_method ->
+            {error, invalid_verification_method};
+        VerificationFn ->
+            {ok, Attr#{verification => VerificationFn}}
+    end.
+
+-spec check_update_method(maybe_module_attribute()) ->
+    {ok, maybe_module_attribute()} | {error, reason()}.
+check_update_method(Attr) ->
+    UpdateMethod = maps:get(update, Attr, read_only),
+    case update_fn(UpdateMethod) of
+        not_exported ->
+            {error, update_method_not_exported};
+        invalid_method ->
+            {error, invalid_update_method};
+        UpdateFn ->
+            {ok, Attr#{update => UpdateFn}}
+    end.
+
+-spec make_module_parameter(maybe_module_attribute(), module()) ->
+    {ok, module_parameter()}.
+make_module_parameter(#{name := Name, default_value := Value, update := UpdateFn,
```

Comment:
Could we use the type `module_attribute()` instead of `maybe_module_attribute()` in this function? It accepts only a specific map as the first argument, not any term.